src/main/java/com/gmail/inverseconduit/utils/ChatBuilder.java

```diff
@@ -0,0 +1,196 @@
+package com.gmail.inverseconduit.utils;
+
+import com.gmail.inverseconduit.datatype.ChatMessage;
+
+/**
+ * Helper class for building chat messages that have SO Chat markdown.
+ * @author Michael Angstadt
+ * @see <a href="http://chat.stackoverflow.com/faq#formatting">Formatting
+ * FAQ</a>
+ */
+public class ChatBuilder implements CharSequence {
```

Comment:
This class has absolutely no validations inbuilt. It is possible to create a tag from "trash tag breaker" and the markdown would bork. For now this is definitely good enough, though ;)